src/lib/simulation/simulation.ts

```diff
@@ -0,0 +1,164 @@
+import {
+    Arc,
+    ArcType,
+    NodeElement,
+    PetriNet,
+    Place,
+    PlaceTransitionArc,
+    Transition,
+    TransitionPlaceArc
+} from '../model';
+
+export class Simulation {
+
+    static readonly ARC_ORDER = [
+        ArcType.INHIBITOR,
+        ArcType.READ,
+        ArcType.REGULAR,
+        ArcType.RESET
+    ];
+    private readonly originalModel: PetriNet;
+    private _simulationModel: PetriNet;
+    private dataVariables: Map<string, number>;
+    private inputArcs: Map<string, Array<Arc<NodeElement, NodeElement>>>;
+    private outputArcs: Map<string, Array<Arc<NodeElement, NodeElement>>>;
+    private assignedTasks: Set<string>;
+    private consumedTokens: Map<string, number>;
+
+    constructor(model: PetriNet, dataVariables = new Map<string, number>()) {
+        if (!model) {
+            throw new Error('Model can not be undefined');
+        }
+        this._simulationModel = this.originalModel = model;
+        this.dataVariables = dataVariables;
+        // noinspection DuplicatedCode
+        this.inputArcs = new Map<string, Array<Arc<NodeElement, NodeElement>>>();
```

Comment:
why is this set in constructor if `reset()` is called afterwards?

Reply:
Lint requires that these properties must be assigned in the constructor :man_shrugging: